subprojects/packagefiles/sdl2/meson_options.txt

```diff
@@ -196,3 +196,10 @@ option(
     type : 'boolean',
     description : 'If possible, dlopen dependencies at runtime instead of linking (linux only)'
 )
+
+option(
+    'sdl2main',
+    type : 'boolean',
+    value : 'true',
+    description : 'Disable including the main function helper'
```

Comment:
The description is now misleading. Maybe it should be something like "Inlude the main() function helper."

Reply:
Options that are of type "disable x" are also confusing because they can't easily not lead to double negatives.

Project option names should not be prefixed with the project name. Meson adds that automatically, so this would actually be called `sdl2:sdl2main` when using from the command line. A better name would be something like `with_main`.

Reply:
> Project option names should not be prefixed with the project name. Meson adds that automatically, so this would actually be called `sdl2:sdl2main` when using from the command line. A better name would be something like `with_main`.

Agreed, although in this case it's a bit ambiguous because the sdl2main option enables linking a static library called, literally, `sdl2main`. So I guess I can understand why people may think it could go either way.